compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -945,13 +945,17 @@
         {
             "name": "RequiredParameterNode",
             "base": "ParameterNode",
-            "kind": "REQUIRED_PARAM",
```

Comment:
Do we need to remove the node kind?

Reply:
Looks like we are using two different kinds for the same node.
But that allows someone to have the asterisk-token in the node, but use the `REQUIRED_PARAM` as the node kind, which is not correct. We can avoid that if we introduce a separate node for `INCLUDED_RECORD_PARAM`.

Reply:
I did this because the asterisk is the only additional token in `included record param` when compared with `required param`. yes. It can be done. Isn't it an intentional work done in the parser side?

Reply:
Catch is, with the existence of asterisk, the node becomes a completely different node, with a different node kind. So the issue I mentioned above is there.

The intention of optional fields is to support truly optional fields for a given node. For eg. metadata (documentation and annotations) is an optional node. But the owner doesn't become a different node, even if the metadata is there or not.

Reply:
Changed as we discussed offline. https://github.com/ballerina-platform/ballerina-lang/pull/27662